source/server/options_impl.cc

```diff
@@ -30,6 +31,24 @@
 #endif
 
 namespace Envoy {
+
+const Stats::Tag parseTag(const std::string tag) {
+  const std::vector<std::string> key_value = StringUtil::split(tag, ":");
+  if (key_value.size() != 2) {
+    throw EnvoyException(
+        fmt::format("Invalid form of --stats-tag was given: '{}'. Use 'name:value' format.", tag));
+  }
+  return Stats::Tag{key_value[0], key_value[1]};
+}
+
+const std::vector<Stats::Tag> parseTags(const std::vector<std::string> tags) {
```

Comment:
`const std::Vector<std::String>&`

Reply:
Also, the `const` in the return doesn't do anything, since it's not a reference. This method could be made static.